mail_composer_cc_bcc/models/ir_mail_server.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2024 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import logging
+
+from odoo import models
+
+_logger = logging.getLogger(__name__)
+
+
+class IrMailServer(models.Model):
+    _inherit = "ir.mail_server"
+
+    def _prepare_email_message(self, message, smtp_session):
+        """
+        Define smtp_to based on context instead of To+Cc+Bcc
+        """
```

Comment:
If i choose to add no default user inside Bcc (from company). But then i add them manually within the composer, i got this error: Invalid email address. It is because of the native _prepare_email_message which checks if email address is inside 'email_to', 'email_cc', 'email_bcc'. However, Bcc addresses is empty (inside the headers instead). So smtp_to_list is empty too and the assert smtp_to_list, self.NO_VALID_RECIPIENT returns the error.

I tested with Odoo 17.4 which is not 18 but closely, and i was able to bypass the error by adding my Bcc addresses list inside the message before calling super().

```
# Find the value for 'X-Odoo-Bcc'
x_odoo_bcc_value = next((value for key, value in message._headers if key == 'X-Odoo-Bcc'), None)
# Add Bcc field inside message to pass validation
if x_odoo_bcc_value:
    message['Bcc'] = x_odoo_bcc_value
```

Reply:
Hi @Jerem-b, indeed, thank you so much :pray: 

I just updated the code to include your contribution